AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/VisualEditorActivity.java

```diff
@@ -0,0 +1,518 @@
+        webView.getSettings().setDefaultFontSize(CardAppearance.calculateDynamicFontSize(mCurrentText));
+
+        webView.setNightMode(cardAppearance.isNightMode(), Themes.getCurrentTheme(this));
+    }
+
+
+    private void handleSelectionChanged(SelectionType selectionType) {
+        SelectionType previousSelectionType = this.mSelectionType;
+
+        this.mSelectionType = selectionType;
+        if (selectionType != previousSelectionType) {
+            invalidateOptionsMenu();
+        }
+    }
+
+
+    private boolean setFieldsOnStartup(Bundle savedInstanceState) {
+
+        tryDeserializeSavedState(savedInstanceState);
+        Bundle extras = this.getIntent().getExtras();
+        if (extras == null) {
+            Timber.w("No Extras in Bundle");
+            return false;
+        }
+
+        if (mCurrentText == null) {
+            mCurrentText = mLargeObjectStorage.getSingleInstance(STORAGE_CURRENT_FIELD, extras);
+        }
+        Integer index = (Integer) extras.getSerializable(VisualEditorActivity.EXTRA_FIELD_INDEX);
+
+        if (mFields == null) {
+            this.mFields = mLargeObjectStorage.getSingleInstance (STORAGE_EXTRA_FIELDS, extras);
+        }
+        Long modelId = (Long) extras.getSerializable(VisualEditorActivity.EXTRA_MODEL_ID);
+
+        if (mCurrentText == null) {
+            Timber.w("Failed to find mField");
+            return false;
+        }
+        if (index == null) {
+            Timber.w("Failed to find index");
+            return false;
+        }
+        if (mFields == null) {
+            return false;
+        }
+        if (modelId == null) {
+            return false;
+        }
+
+        this.mModelId = modelId;
+
+        mIndex = index;
+
+        mWebView = this.findViewById(R.id.editor_view);
+
+        if (mWebView == null) {
+            Timber.w("Failed to find WebView");
+            return false;
+        }
+
+        return true;
+    }
+
+
+    private void tryDeserializeSavedState(@Nullable Bundle savedInstanceState) {
+        if (savedInstanceState == null) {
+            return;
+        }
+        if (mLargeObjectStorage.hasKey(STORAGE_CURRENT_FIELD, savedInstanceState)) {
+            mCurrentText = mLargeObjectStorage.getSingleInstance(STORAGE_CURRENT_FIELD, savedInstanceState);
+        }
+        if (mLargeObjectStorage.hasKey(STORAGE_EXTRA_FIELDS, savedInstanceState)) {
+            mFields = mLargeObjectStorage.getSingleInstance(STORAGE_EXTRA_FIELDS, savedInstanceState);
+        }
+    }
+
+
+    private void failStartingVisualEditor() {
+        UIUtils.showThemedToast(this, "Unable to start visual editor", false);
+        finishCancel();
+    }
+
+
+    @Override
+    protected void onCollectionLoaded(Collection col) {
+        super.onCollectionLoaded(col);
+        Timber.d("onCollectionLoaded");
+        try {
+            initWebView(col);
+        } catch (IOException e) {
+            Timber.e(e, "Failed to init web view");
```

Comment:
Change all the `Timber.w` to `Timber.e`

```suggestion
            Timber.w(e, "Failed to init web view");
```

Reply:
it confusing,

in comment, you said "change Timber.w to Timber.e"

and in suggested code you changed "Timber.e to Timber.w"

Reply:
I profusely apologise. `Timber.w` for all but non-fatal errors

Reply:
I profusely apologise, I'd made the correct suggestion, and the incorrect comment